### PR TITLE
ADD: Add CMake build option for tests and example C++ program

### PR DESCRIPTION
**Pull request by T-K-233**
Add support to build tests with CMake:

To build the programs, do

```bash
# cd ./tests/
cmake . -S ./ -B ./build/ -D CMAKE_BUILD_TYPE=Debug
cmake --build ./build/ --target all
```

To clean the previous build, do

```bash
# cd ./tests/
cmake --build ./build/ --target clean
```

**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [x] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [x] Software change (RISC-V software)
- [x] Build system change
- [ ] Other

**Contributor Checklist**:
- [x] Did you set `main` as the base branch?
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you state the type-of-change/impact?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

**CI Help**:
Add the following labels to modify the CI for a set of features.
Generally, a label added only affect subsequent changes to the PR (i.e. new commits, force pushing, closing/reopening).
See `ci:*` for full list of labels:
- `ci:fpga-deploy` - Run FPGA-based E2E testing
- `ci:local-fpga-buildbitstream-deploy` - Build local FPGA bitstreams for platforms that are released
- `ci:disable` - Disable CI

### CODE REVIEW

**Comment by jerryz123** [comment]
All the C programs build with this too? That's neat.  

Can you replace the existing makefile such that 'make' just invokes cmake?

**Comment by T-K-233** [comment]
I don't think using make to invoke cmake is a good idea... It's a bit convoluted.
If we want to use cmake to *replace* the current make flow, I can write a .sh script to abstract away the cmake commands, so user would just need to call something like `./build.sh`.

**Comment by jerryz123** [comment]
Replacing it with build.sh is fine to me too. Note that many parts of the CI scripts call make, those would have to be changed to build.sh

File: scripts/build-tests.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+cmake ./tests/ -S ./tests/ -B ./tests/build/ -D CMAKE_BUILD_TYPE=Debug
```
Comment (jerryz123) [inline scripts/build-tests.sh:1]: Add a `#!/bin/bash` to the head of this script.

File: .github/scripts/run-tests.sh [hunk 2]
```diff
@@ -89,29 +93,41 @@ case $1 in
...
-	make -C $LOCAL_CHIPYARD_DIR/tests
+	    cmake $LOCAL_CHIPYARD_DIR/tests/ -S $LOCAL_CHIPYARD_DIR/tests/ -B $LOCAL_CHIPYARD_DIR/tests/build/ -D CMAKE_BUILD_TYPE=Debug
+        cmake --build $LOCAL_CHIPYARD_DIR/tests/build/ --target all
+
         run_binary BINARY=$LOCAL_CHIPYARD_DIR/tests/nvdla.riscv LOADMEM=1
 	;;
     chipyard-manyperipherals)
 	# SPI Flash read tests
-        make -C $LOCAL_CHIPYARD_DIR/tests
+        cmake $LOCAL_CHIPYARD_DIR/tests/ -S $LOCAL_CHIPYARD_DIR/tests/ -B $LOCAL_CHIPYARD_DIR/tests/build/ -D CMAKE_BUILD_TYPE=Debug
```
[the remark this answers is not in the document]
Comment (T-K-233) [reply .github/scripts/run-tests.sh:103]: what does DRY mean?

[the remark this answers is not in the document]
Comment (jerryz123) [reply .github/scripts/run-tests.sh:103]: "Don't repeat yourself". It means just avoiding duplicate code or commands.

**Review by jerryz123 — APPROVED** [review]

**Pull request merged**